mmdet/models/dense_heads/detr_head.py

```diff
@@ -833,8 +833,11 @@ def onnx_export(self, all_cls_scores_list, all_bbox_preds_list, img_metas):
         img_shape_tensor = img_shape_tensor.unsqueeze(0).unsqueeze(0).expand(
             batch_size, det_bboxes.size(1), 4)
         det_bboxes = det_bboxes * img_shape_tensor
-        det_bboxes[..., 0::2].clamp_(min=0, max=img_shape[1])
-        det_bboxes[..., 1::2].clamp_(min=0, max=img_shape[0])
+        # dynamically clip bboxes
+        x1, y1, x2, y2 = det_bboxes.split((1, 1, 1, 1), dim=-1)
+        from mmdet.core.export import dynamic_clip_for_onnx
+        x1, y1, x2, y2 = dynamic_clip_for_onnx(x1, y1, x2, y2, img_shape)
+        det_bboxes = torch.cat([x1, y1, x2, y2], dim=-1)
```

Comment:
What's the performance after this change? Is it aligned with batchsize==1?

Reply:
40.1 mAP with batch_size 1 and 33.6 mAP with batch_size 2, after the modification.

Reply:
Then, we may need to find real causes for the performance drop with batch-size > 1.

Reply:
As I mentioned at the beginning, there are two factors affect the performance of torch and onnx model, the second is the main.

Reply:
@v-qjqs 
Then, is there a constant shape that images from a batch should be padded into in the original PyTorch inference pipeline?

Reply:
Actually the padding for images for constructing a batch tensor is done by mmcv, and the `size_divisor` in DETR is set to 1.